client/src/js/components/bhCurrencySelect.js

```diff
@@ -97,8 +97,6 @@ function bhCurrencySelect($scope, Currencies) {
 
     // if the two array lengths are equal, it means every currency is disabled
     $ctrl.allDisabled = ($ctrl.currencies.length === array.length);
-    if ($ctrl.allDisabled) {
-      $ctrl.form.$setValidity('currencies', false);
-    }
+    $ctrl.form.currency.$setValidity('currency', !$ctrl.allDisabled);
```

Comment:
Yes, but I should probably alias this with a local variable.    Something like:
```js
$ctrl.valid = $ctrl.currencies.length === array.length;

$ctrl.form.currency.$setValidity('currency', $ctrl.valid);
```

The problem with the previous implementation is that it needs to be added and removed dynamically - so setting it to false would never reset it to true.  This was a pretty big bug.

I'll make the changes before merge.